### PR TITLE
Teacher student show

### PR DESCRIPTION
**Pull request by Arique1104**
## Pull request type
Please check the type of change your PR introduces:

- [x ] Bugfix
- [ x] Feature
- [ x] Code style update (formatting, renaming)
- [ x] Refactoring (no functional changes, no api changes)
- [ x] Build related changes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please describe):

## Problem/feature

- Wish we could've broken this up, but this pull request is ultimately working towards the course show page in teachers and students is working.  
- Updates for all oauth handshaking to work.
- Updates for all current user including the distinction between student and teacher.
- All teachers/courses endpoints are exposed, tested, and working
- Same with students/courses 

## Solution

Group code sharing was what helped this really come through, that and @priyapower's amazing leadership.

## Other changes (e.g. bug fixes, UI tweaks, small refactors)

Run the rake test BEFORE running your rails s:

```console
rake csv_import:seedCSV
```

## Checklist
- [ x] Code compiles correctly
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x ] All tests passing
- [ ] No rubocop offenses
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by sarynm12 — APPROVED** [review]
Amazing work :)

File: app/controllers/api/v1/students_controller.rb [hunk 1]
```diff
@@ -12,11 +12,16 @@ def show
       end
 
       def create
-        Student.create(student_params)
-        render json: StudentSerializer.new(Student.last)
+        # Student.create(student_params)
+        # render json: StudentSerializer.new(Student.last)
```
Comment (fieldstyler) [inline app/controllers/api/v1/students_controller.rb]: Are we saving this for later or can it be removed?

Comment (priyapower) [reply app/controllers/api/v1/students_controller.rb]: Uh oh! This can be removed

File: app/controllers/api/v1/teachers_controller.rb [hunk 2]
```diff
@@ -10,11 +10,16 @@ def show
       end
 
       def create
-        Teacher.create(teacher_params)
-        render json: TeacherSerializer.new(Teacher.last)
+        # Teacher.create(teacher_params)
+        # render json: TeacherSerializer.new(Teacher.last)
```
Comment (fieldstyler) [inline app/controllers/api/v1/teachers_controller.rb]: Same question. Are we leaving this for a reason?

Comment (priyapower) [reply app/controllers/api/v1/teachers_controller.rb]: They are gone now!

File: app/models/war.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class War < ApplicationRecord
+  belongs_to :teacher
+
+  validates_presence_of :challenger_course_id,
+                        :opponent_course_id,
+                        :challenger_course_points,
+                        :opponent_course_points
+
```
Comment (fieldstyler) [inline app/models/war.rb:8]: what's the difference between challenger and opponent?

File: spec/models/course_spec.rb [hunk 4]
```diff
@@ -5,12 +5,13 @@
...
-    it { should validate_presence_of :course_code }
     it { should validate_presence_of :school_name }
     it { should validate_presence_of :course_points}
   end
 
   describe 'Relationships' do
     it { should belong_to :teacher }
   end
+
+  # NEEDS A TEST FOR COURSE MODEL METHOD GENERATE CODE
```
Comment (fieldstyler) [inline spec/models/course_spec.rb:16]: Do we have a plan for this?

**Comment by fieldstyler** [comment]
Reviewed by: Tyler Fields

**Pull request merged**